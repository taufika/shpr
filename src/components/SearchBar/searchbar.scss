@import '../../variables/colors.scss';
@import '../../variables/mixins.scss';

.search-bar {
  background-color: #fff;
  padding: 20px;
  display: flex;
  align-items: center;
  justify-content: space-between;

  h1 {
    color: $primary-color;
    margin-top: 0;
    margin-bottom: 4px;
  }

  h3 {
    margin: 0;
    font-size: 12px;
    font-weight: normal;
  }

  .search-area {
    display: flex;

    .search-icon {
      border: 1px solid #727272;
      border-right: 0;
      padding: 8px 12px;
      color: $primary-color;
    }

    input {
      border-radius: 0;
      border: 1px solid #727272;
      border-left: 0;
      padding: 8px 12px;
      padding-left: 0;
      width: 140px;
      margin-right: 12px;
      outline: none;
    }
  }

  button.add-driver {
    border-radius: 0;
    border: 1px solid $primary-color;
    background-color: $primary-color;
    padding: 8px 12px;
    color: #fff;

    .fas {
      margin-left: 12px;
    }
  }
}

@include small-mobile-view() {
  .search-bar {
    display: block;

    .title-area {
      margin-bottom: 20px;
    }

    .search-area {
      flex-wrap: wrap;
      row-gap: 20px;

      input {
        width: calc(100% - 54px);
        margin-right: 0;
      }

      button.add-driver {
        width: 100%;
      }
    }
  }
}
