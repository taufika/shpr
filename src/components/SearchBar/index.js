import React from 'react';

import './searchbar.scss';

export default class SearchBar extends React.Component {
  handleInput(e) {
    this.props.keywordHandler(e.target.value);
  }

  render() {
    return <div className='search-bar'>
      <div className='title-area'>
        <h1>Driver Management</h1>
        <h3>Data driver yang bekerja dengan Anda.</h3>
      </div>
      <div className='search-area'>
        <div className='search-icon'>
          <i className="fas fa-search"></i>
        </div>
        <input value={this.props.keyword} placeholder='Cari Driver' onChange={this.handleInput.bind(this)}/>
        <button>
          Tambah Driver
          <i className="fas fa-plus"></i>
        </button>
      </div>
    </div>;
  };
}