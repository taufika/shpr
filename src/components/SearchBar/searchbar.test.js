import { render, screen, fireEvent } from '@testing-library/react';
import SearchBar from '.';

describe('Components:SearchBar', () => {
  it('Should render and match snapshot', () => {
    const { container } = render(<SearchBar />);
    expect(container).toMatchSnapshot();
  });

  it('Should be able to accept keyword props', () => {
    render(<SearchBar keyword='abcdefg' />);
    const input = screen.getByTestId('search-input');
    expect(input.value).toBe('abcdefg');
  });

  it('Should handle input change', () => {
    const listener = jest.fn();
    render(<SearchBar keywordHandler={listener} />);
    const input = screen.getByTestId('search-input');
    fireEvent.change(input, { target: { value: 'new keyword' } });
    expect(listener).toHaveBeenCalledWith('new keyword');
  });
});
