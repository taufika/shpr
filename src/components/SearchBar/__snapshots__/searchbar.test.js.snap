// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components:SearchBar Should render and match snapshot 1`] = `
<div>
  <div
    class="search-bar"
  >
    <div
      class="title-area"
    >
      <h1>
        Driver Management
      </h1>
      <h3>
        Data driver yang bekerja dengan Anda.
      </h3>
    </div>
    <div
      class="search-area"
    >
      <div
        class="search-icon"
      >
        <i
          class="fas fa-search"
        />
      </div>
      <input
        data-testid="search-input"
        placeholder="Cari Driver"
        value=""
      />
      <button
        class="add-driver"
      >
        Tambah Driver
        <i
          class="fas fa-plus"
        />
      </button>
    </div>
  </div>
</div>
`;
